buildlib/test.sh

```diff
@@ -0,0 +1,202 @@
+
+NODELIST=${NODELIST:="localhost"}
+
+UCX_LIB=${UCX_LIB:=${LD_LIBRARY_PATH}}
+
+SPARK_UCX_JAR=${SPARK_UCX_JAR:=$PWD/spark-ucx-1.0-for-spark-2.4-jar-with-dependencies.jar}
+
+PROCESSES_PER_INSTANCE=${PROCESSES_PER_INSTANCE:=2}
+
+if [ -n "$EXECUTOR_NUMBER" ] 
+then
+	AFFINITY="taskset -c $(( 2 * EXECUTOR_NUMBER ))","$(( 2 * EXECUTOR_NUMBER + 1))"
+else
+	AFFINITY=""
+fi
+
+EXECUTOR_NUMBER=${EXECUTOR_NUMBER:=$((RANDOM % `nproc`))}
+
+JOB_ID=${SLURM_JOB_ID:="sparkucx-test-$(cat /proc/sys/kernel/random/uuid)"}
+
+SCRATCH_DIRECTORY=${SCRATCH_DIRECTORY:=$PWD/${JOB_ID}}
+
+SPARK_LOCAL_DIRS=${SPARK_LOCAL_DIRS:="/scrap/sparkucxtest/sparkucx-${JOB_ID}"}
+
+SPARK_WORKER_CORES=${SPARK_WORKER_CORES:=2}
+
+SPARK_WORKER_MEMORY=${SPARK_WORKER_MEMORY:="1g"}
+
+SPARK_HOME=${SPARK_HOME:=$PWD/spark}
+
+SPARK_VERSION=${SPARK_VERSION:="2.4.4"}
+
+UCX_BRANCH=${UCX_BRANCH:="v1.7.x"}
+
+export SPARK_MASTER_HOST=${SPARK_MASTER_HOST:=$(hostname -f)}
+export SPARK_MASTER_PORT=${SPARK_MASTER_PORT:=$(( 2000 + 10 * ${EXECUTOR_NUMBER}))}
+export SPARK_CONF_DIR=${SCRATCH_DIRECTORY}/conf
+
+get_rdma_device_iface() {
+
+	if [ ! -r /dev/infiniband/rdma_cm  ]
+	then
+		return
+	fi
+
+	if ! which ibdev2netdev >&/dev/null
+	then
+		return
+	fi
+
+	iface=`ibdev2netdev | grep Up | awk '{print $5}' | head -1`
+	if [[ -n "$iface" ]]
+	then
+		ipaddr=$(ip addr show ${iface} | awk '/inet /{print $2}' | awk -F '/' '{print $1}')
+	fi
+
+	if [[ -z "$ipaddr" ]]
+	then
+		# if there is no inet (IPv4) address, escape
+		return
+	fi
+
+	ibdev=`ibdev2netdev | grep ${iface} | awk '{print $1}'`
+	node_guid=`cat /sys/class/infiniband/$ibdev/node_guid`
+	if [[ ${node_guid} == "0000:0000:0000:0000" ]]
+	then
+		return
+	fi
+
+	SPARK_MASTER_HOST=$ipaddr
+	echo ${iface}
+}
+
+RDMA_NET_IFACE=${RDMA_NET_IFACE:=`get_rdma_device_iface`}
+
+download_spark() {
+    curl -s -L -O https://www-eu.apache.org/dist/spark/spark-${SPARK_VERSION}/spark-${SPARK_VERSION}-bin-hadoop2.7.tgz
+    mkdir -p ${SPARK_HOME}
+    tar -xf spark-${SPARK_VERSION}-bin-hadoop2.7.tgz -C ${SPARK_HOME} --strip-components=1
+}
+
+build_ucx() {
+    git clone -b ${UCX_BRANCH} https://github.com/openucx/ucx.git && cd  ucx
+    ./autogen.sh
+    mkdir build && cd build
+    ../contrib/configure-release-mt --with-java --prefix=$PWD
+    make -j `nproc`
+    make install
+    UCX_LIB=$PWD/lib/
+}
+
+setup_configuration() {
+    mkdir -p ${SPARK_CONF_DIR}
+
+    echo ${NODELIST} | tr -s ' ' '\n' >> "${SPARK_CONF_DIR}/slaves"
+
+    cat <<-EOF > ${SPARK_CONF_DIR}/spark-defaults.conf
+	spark.shuffle.manager org.apache.spark.shuffle.UcxShuffleManager
+	spark.driver.extraClassPath ${SPARK_UCX_JAR}:${UCX_LIB}
+	spark.executor.extraClassPath ${SPARK_UCX_JAR}:${UCX_LIB}
+	spark.shuffle.ucx.driver.port $(( ${SPARK_MASTER_PORT} + 1 ))
+	EOF
+
+    cat <<-EOF > ${SPARK_CONF_DIR}/spark-env.sh
+	export SPARK_LOCAL_IP=\`/sbin/ip addr show ${RDMA_NET_IFACE} | grep "inet\b" | awk '{print \$2}' | cut -d/ -f1\`
```

Comment:
use helper variable SPARK_IP_ADDR or similar

Reply:
This is run on every node to determine each executor ip address. It's a script, not actual variable - it would be different on each node.